app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,185 @@
+from django.conf import settings
+from django.core.paginator import Paginator
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+from data.models import (DedupeJob, RecordDuplicate)
+from ashlar.models import Record
+
+import datetime
+from pytz import timezone
+
+logger = get_task_logger(__name__)
+
+
+def get_time_allowance():
+    return datetime.timedelta(hours=settings.DEDUPE_TIME_RANGE_HOURS)
+
+
+# .001 degrees is ~110m give or take 10 depending on location, which should
+# be good enough
+def get_distance_allowance():
+    return settings.DEDUPE_DISTANCE_DEGREES
+
+
+def calculate_score(record, duplicate):
+    """
+    Set a score representing the likelyhood the two records are duplicated, based
+    on the criteria.  score is a float between 0 and 1
+
+    normalized dimensions are considered equal (1 day equally as important as 100m)
+    """
+    tdelta = abs(record.occurred_from - duplicate.occurred_from)
+    tscore = 0
+    if tdelta.total_seconds() == 0:
+        tscore = 1.0
+    else:
+        tscore = tdelta.total_seconds() / get_time_allowance().total_seconds()
+    ddelta = record.geom.distance(duplicate.geom)
+    dscore = 0
+    if ddelta == 0.0:
+        dscore = 1.0
+    else:
+        dscore = ddelta / get_distance_allowance()
+    score = (tscore + dscore) / 2
+    return score
+
+
+def find_duplicates_for_record(uuid):
+    """ Given a record, find all possible duplicates for it
+    """
+    record = Record.objects.get(uuid=uuid)
+    return Record.objects.filter(
+        schema__record_type=record.schema.record_type,
+        geom__dwithin=(record.geom, get_distance_allowance()),
+        occurred_from__range=(
+            record.occurred_from - get_time_allowance(), record.occurred_from + get_time_allowance()
+        )
+    ).exclude(uuid=record.uuid)
+
+
+def get_time_extent():
+    """ Get the time range to dedupe
+    Calculate from X hr before the last dedupe run to the current task run's
+    start time, where X is the heuristic's time allowance
+
+    Returns:
+        dictionary containing the start and end times for the next dedupe task
+        {starttime: timestamp, endtime: timestamp}
+    """
+    # TODO: get config var determining matching time - use value of 1 hr for now
+    time_allowance = get_time_allowance()
```

Comment:
I might be misunderstanding, but I'm not sure we should use the same time allowance for both the time range to dedupe over and for determining the similarity between records; these two things may have different sweet spots in terms of performance vs. false positives so we may want to set them to different values.